components/homepage/HomeScreen.js

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
+import { List, Checkbox } from 'react-native-paper';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import { useSelector, useDispatch } from 'react-redux';
+import { createSelector } from '@reduxjs/toolkit';
+import { View, Text, Button, TouchableOpacity, ScrollView } from 'react-native';
+import {
+  fetchAllTasks,
+  updateTasks,
+  setTaskStatus,
+  fetchAllTmrTasks,
+  updateTmrTasks,
+  setTmrTaskStatus,
+} from '../../redux/TasksSlice';
+
+import {
+  Bars3CenterLeftIcon,
+  PlusCircleIcon,
+} from 'react-native-heroicons/solid';
+import { colors } from '../../theme/colors';
+import { formatTime } from '../../helper/TimeFormat';
+
+const HomeScreen = ({ navigation }) => {
+  const dispatch = useDispatch();
+
+  const tasksSelector = createSelector(
+    state => state.tasks,
+    tasks => {
+      return tasks.listTasks;
+    }
+  );
+
+  const tasks = useSelector(tasksSelector);
+
+  const tmrTasksSelector = createSelector(
+    state => state.tasks,
+    tasks => {
+      return tasks.listTmrTasks;
+    }
+  );
+
+  const tmrTasks = useSelector(tmrTasksSelector);
+
+  const markDoneTask = taskId => {
+    dispatch(setTaskStatus({ taskId, isToday: true }));
```

Comment:
mark task done should just need `taskId` to update `isDone` attribute to true. I'm not sure why do we need `isToday`? We shouldn't use `isToday: true` to check it's today tasks. We should have a `createdAt` attribute instead. With `createdAt` we can compare with the current date, if they are the same date mean today tasks.